diff --git a/include/spdlog/details/os.h b/include/spdlog/details/os.h
index 735f601..6241aa5 100644
--- a/include/spdlog/details/os.h
+++ b/include/spdlog/details/os.h
@@ -342,7 +342,7 @@ inline size_t _thread_id()
 //Return current thread id as size_t (from thread local storage)
 inline size_t thread_id()
 {
-#if defined(_MSC_VER) && (_MSC_VER < 1900) || defined(__clang__) && !__has_feature(cxx_thread_local)
+#if !defined(LIZARDFS_HAVE_THREAD_LOCAL) || defined(_MSC_VER) && (_MSC_VER < 1900) || defined(__clang__) && !__has_feature(cxx_thread_local)
     return _thread_id();
 #else
     static thread_local const size_t tid = _thread_id();
